### ISSUE TITLE
App Development Discussion 

### ISSUE THREAD

**Issue by bdshanks-28** [IanTDuncan/MealTime#60]
App Development Discussion
---------------------------------------------------

Description:
------------------------------

This is the issue created for our online/in-person meeting notes; general discussion, questions and decisions that have been made.

Issue Status: Open; In-Progress
------------------------------

**Comment by aaleksandraristic** [comment]
From our last roadmap:
------------------------------------------------
Month 4-5: Define MVP Features (Minimum Viable Product); UI/UX Design
Prioritize features based on user feedback and business objectives
Create a feature list for the Minimum Viable Product (MVP), including food analysis and nutrient tracking
Develop wireframes and prototypes
Design the app's user interface, including screens for grocery list creation and meal generation
Conduct usability testing
Create design specifications for development

Quarter 2: App Development
Month 5-8: MVP Development
Develop the core functionality of the app, including food analysis, nutrient tracking, and calorie counting
Implement food database integration for nutrient data
Create user registration and authentication systems

Month 8-9: Testing and Quality Assurance
Conduct rigorous testing, including functional, performance, and security testing
Identify and address any bugs or issues

**Quarter 3: Launch **
-> Beta Testing
Invite a small group of users to participate in beta testing, including testing of grocery list and meal generation features
Gather feedback and make necessary improvements

Month 10: App Launch
Publish the MealTime app with all planned features on app stores (iOS and Android)
Implement a marketing and promotion plan to attract users
Monitor user feedback and app performance

**Comment by aaleksandraristic** [comment]
Visualised roadmap:
--------------------------------

https://miro.com/app/board/uXjVNUQNtuM=/

**Comment by aaleksandraristic** [comment]
Let's take a closer look today and decide what would be some other things that need to be added, as well as a timeframe for all of this.
@BeepDroid @Cade5480 @bdshanks-28 @IanTDuncan

**Comment by Cade5480** [comment]
Sounds good, excited to get back going!!

**Comment by bdshanks-28** [comment]
Going to use discord for voice calls and meetings. At least one person will be recording to then post the recording to github for reviewing.

**Comment by bdshanks-28** [comment]
Android studio installed and configured!

**Comment by BeepDroid** [comment]
Same here! Working on getting the design converted as we speak, need to add the button functionality since I've confirmed converting is possible.

**Comment by BeepDroid** [comment]
Okay, so I have the APK file and have run it through Android Studios and have it compiled with a zip file. I have some things to fix configuration wise, and I know I already mentioned this in Slack but if anyone else can test it with their Studio emulated device to make extra sure it works that would be awesome. 

Unfortunately, ran into a problem uploading the files to the repository -- they're too big! I looked into it and I think I found a way around it using GitLFS that's specific for this problem. 

Anyway, I still can't attach it here in issues, so I'm gonna send the folder through Slack. Maybe it'll make it easier for you guys instead of having to go through and test the APK file all over again (it took some time).

**Comment by bdshanks-28** [comment]
[Documentation](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github) for file sizes on github. So browser uploads cap at 25 mb, they don't like 50 mb uploads, and 100 mb uploads do just get blocked. Hope this helps us out.

**Comment by BeepDroid** [comment]
I got it commited! Accidentally committed both the zip version and full, so I'll get rid of that zip folder so it doesn't eat up more room. But it did help, thank you so much! The console version took it much easier.

**Comment by aaleksandraristic** [comment]
I will write here our tasks once again, so we can have it saved here:
----------------------------------------------------------------------------------
(We are going to do color coding for all the issues that you guys create while working on these tasks so it will be easier to manage all the issues later on)
Database - Alex (BLUE)
Data privacy - Kerry (RED)
Methods/Classes - Cade (AQUA)
APIs - Ian
UI/UX - Brandon

Firebase - ON HOLD FOR NOW

**Comment by aaleksandraristic** [comment]
We have met on January 28th (today) and discussed about what have been achieved and what were some problems. We were all working on our tasks in person together so we were able to talk to each other while working. Our meeting lasted from 1pm until 4pm. 

If all of us could please write below what have been achieved so far (for the whole month) and what were some problems that you faced while working on your tasks so we can summarize it for the retrospective issue which is due on Wednesday? 

Thank you all for meeting today! 
@BeepDroid @Cade5480 @bdshanks-28 @IanTDuncan

**Comment by bdshanks-28** [comment]
I worked on smoothing out the login page as well as adding a list for use when the bookmarks page gets more data into it.

**Comment by BeepDroid** [comment]
I worked on more of the Privacy Policy and began the beginning tabs of information for our Terms of Service!

**Comment by IanTDuncan** [comment]
I worked on implementing the Edamam food database for our app. However, I left it incomplete due to issues understanding the JSON file and decided to pivot to working on implementing the Edmamam Recipe Search.

**Comment by aaleksandraristic** [comment]
Potential problems that we had:
- At the beginning of the semester the leader of the team (me) didn't organize the team's work well. The way how we solved it is by asking a professor about the main points of the app development. After I was advised, I split the tasks between team members. 

(this will be part of the retrospective at the end of he milestone)

**Comment by aaleksandraristic** [comment]
![image](https://github.com/IanTDuncan/MealTime/assets/140200824/ca189347-6f6a-4d92-b907-c9401e09d3dc)

**Comment by aaleksandraristic** [comment]
Okay guys, here are some new things that would be good to follow from now on. He talked to us about how to be better, and here is what we need to change:

- everything that we do needs to be separated into detailed issues and described exactly what is about
- we all are going to create our own issues for each part (as Brandon did for UI/UX)
- after that, I would have to sit down with all of you separately and discuss the timeline and the plan for when and what should be done first, second, and so on
- to make it easier to track, I would recommend using the Project feature that Ian created inside of our GitHub repository
- whenever something passes the deadline or make a mistake, I have to write it down in retrospective and call you out in front of everyone

I would put all of this like steps but are just main things that should be done better than we do rn:
-----------------------------------------------------------------------------------------------------------------
The main thing right now is to create all the issues and tasks inside of the [project](https://github.com/users/IanTDuncan/projects/1).
Meet me (online / in-person / text) and let me know what is what and how much time you expect to spend on it
Then we will set what should be done first and set a deadline
Decide what the following tasks and app deadlines
From now on everything should be commented out in GitHub issues feel free to put a copy of the code over there and shortly explain what you did, and how's going (even if it's not working - some of us might be able to help)
When we do standups, everything has to be more specific, the exact point of what you did and what is next
At the end retrospective should be more specific and I have to call out your name if you were stuck on some issue or you had any problem and then talk to you about it
(like Aleksandra had a problem with committing things while she was working on this and this issue and here is how she solved it)
 8. Feel free to write / take notes on all the problems or what went wrong with your issues and tag me as well so I would be aware of what is going on
 
 -------------------------------------------------------------------------
I hope this is not too stressful!! We just need to pass this tough period and we'll be fine! I am very proud of y'all and very sorry that I am not leading you well, but here we are to learn and improve ourselves. Please let me know your thoughts and suggestions, everything would help.
@BeepDroid @Cade5480 @bdshanks-28 @IanTDuncan

**Comment by aaleksandraristic** [comment]
I have changed the name of this issue since it's going to last more than a January and we will use it as we did, for discussion and planning.

**Comment by aaleksandraristic** [comment]
I have talked to my team and we are planning to work on above changes and organization for our next standup.
Ideally, we would be able to finish some of our issues, but the main thing right now is making a good plan and timeline.

Thank you guys!
@BeepDroid @Cade5480 @bdshanks-28 @IanTDuncan 

(I'll keep everything that we decided here, so we have a note what we have planned)

**Comment by aaleksandraristic** [comment]
Everyone contributed to the changes and we came up with first plan for all the issues and its due dates

**Comment by BeepDroid** [comment]
Meeting notes regarding this issue (Meeting Date: 2/12/2024):

- Cade and I reviewed the Terms and Conditions document together and discussed formatting possiblitles.
- For the backend development, we agreed that this particular document could be accessible through the "About" section in the Accounts page on the App.
- The access would be through a PDF link to the aforementioned document, which we did discover you could download as is from the Google doc file.

- Cade and I reviewed Privacy Policy document, typos fixed, and it is now ready for inclusion.
- Logic regarding this document was discussed as being the first pop-up upon access, with a check event at the bottom to show the user has read and agreed that they understood the clauses listed in the policy.

Reposting of meeting notes from other issues for @aaleksandraristic easy access!

**Comment by aaleksandraristic** [comment]
Thank you Kerry for providing us with these notes, I appreciate it!

**Comment by bdshanks-28** [comment]
I am not going to create any new issues. Instead I want to clean up the xml pages I have before making new issues since I might have redundancy.

**Comment by bdshanks-28** [comment]
Hey so I'm putting this here for now but I am gonna try and help Cade out with reorganizing the xml files and buttons pointing to the right ones so I can go and delete some as part of a cleanup effort. I attached the main overview to issue #65 for @Cade5480 to look over.

@aaleksandraristic I'm including you as well just so you are aware.

**Comment by aaleksandraristic** [comment]
Okay, that sounds good, thank you for letting me know!

**Comment by IanTDuncan** [comment]
### Questions From Kerry

1. Can you give me an overview about how Firebase database works in terms of taking information and sending queries? I want to start coding the classes for the Shopping List and I know that data is going to be stored, so I was wondering if the database could take in a list of data points or if it has to be segmented out?
2.If the answer to the question above is no, it cannot take data as a list, could it send out data into a List object so it can be viewed as such on the XML page?
3. A general question, given his feedback from last class, should I make a new issue for the class I'm writing or just post the code in the Shopping List API issue I've already made? OR should I just use that issue for research and keep my code to whatever class issues I need to make? (I'm sorry if this is confusing).


### Question(s) From Ian

1. For tracking the recipes for Meal Planner by user, I was thinking of using URIs, a **Uniform Resource Identifier**, this is already being used in by Meal Planner, and we can store with User IDs in the Firebase database?

**Comment by bdshanks-28** [comment]
I think an answer for 2 @BeepDroid would be yes to be on the safe side. That way we know what is correctly added and what it does without digging through the whole issue.

**Comment by aaleksandraristic** [comment]
1. I would need to check it once again and see how to explain it, is it okay if I answer this one later today? 

2. Yeah, I agree with Brandon, let’s create a new one so we can be safe in that way.

**Comment by aaleksandraristic** [comment]
@IanTDuncan please correct me if I am wrong, so we will store Uniform Resource Identifier with User ID so we can store the data for tracking recepies for each user, right?

**Comment by IanTDuncan** [comment]
@aaleksandraristic I believe you will have to make a new table to track the URIs. My suggestions would be to have a table composed of the User ID and track the URIs by food the meal types shown here: 
![Screenshot 2024-02-25 130753](https://github.com/IanTDuncan/MealTime/assets/143751333/8bdfb27f-e705-4793-be2b-758d2c6440fa)

**Comment by aaleksandraristic** [comment]
I guess we can close this issue, since everything here is related to January and Februaru. Let's use new issue -[ MVP #127](https://github.com/IanTDuncan/MealTime/issues/127) for our further communication.
